src/microbe_stage/Spawners.cs

```diff
@@ -321,6 +322,20 @@ public MicrobeSpawner(Species species, CompoundCloudSystem cloudSystem, GameProp
 
     public override IEnumerable<ISpawned> Spawn(Node worldNode, Vector3 location)
     {
+        var microbes = worldNode.GetTree()
+            .GetNodesInGroup(Constants.AI_TAG_MICROBE)
+            .Cast<Microbe>()
+            .ToList();
+
+        var microbesCount = microbes.Count;
+        var speciesMemberCount = microbes.Count(p => p.Species == species);
+
+        if (speciesMemberCount >= Constants.MAX_CELLS_OF_SPECIES_ON_SCREEN)
+            yield break;
+
+        if (microbesCount >= Constants.MAX_CELLS_ON_SCREEN)
+            yield break;
+
```

Comment:
Instead of specialist logic like this, the spawn system should be configured to allow a cap per spawn type (then the patch manager or something can allocate the caps for each species depending on their populations)

Reply:
Do you mean I should add a `boolean LimitReached()` method in `Spawner`?

Reply:
Basically, yes, but it shouldn't be a method, instead it should be a number that the spawn system takes care of maintaining.

Reply:
So I should save the number of how many spawned entities of each type are on screen at the moment.
Is there an event or something like that that notifies me whenever something despawns?

Reply:
Arguably there is already the despawn check in the spawn system, that could be extended to keep track of how many alive entities each spawner has.